### ISSUE TITLE
Since 5.0.2 radio streaming doesn't work / remote control

### ISSUE THREAD

**Issue by MasteRehm** [OpenELEC/OpenELEC.tv#3900]
Hi,

about 2 hours ago I updated from 5.0.1 to 5.0.2 and I recognize radio streaming via add-on radio [1] doesn't work anymore. In rare cases a selected station is playing, but, and here it's getting weird, my remote control app Yatse doesn't work properly anymore. The app says Kodi (OpenELEC) is offline, but only as long as the radio station is playing.
So, what did I do? I simply downgraded to Version 5.0.1 and everything works as expected again.

Here some kodi.log entries from Ver. 5.0.2, right after I selected a station
```
20:36:09 T:1655989312  NOTICE: -->Python Interpreter Initialized<--
20:36:10 T:1655989312  NOTICE: [xbmcswift2] Request for "/station/1623" matches rule for function "get_stream_url"
20:36:10 T:1655989312  NOTICE: [plugin.audio.radio_de] get_station_by_station_id started with station_id=1623
20:36:10 T:1655989312  NOTICE: [plugin.audio.radio_de] __api_call started with path=broadcast/getbroadcastembedded, param={'broadcast': '1623'}
20:36:10 T:1655989312  NOTICE: [plugin.audio.radio_de] __urlopen opening url=http://radio.de/info/broadcast/getbroadcastembedded?broadcast=1623
20:36:10 T:1655989312  NOTICE: [plugin.audio.radio_de] __resolve_playlist started with station=1623
20:36:10 T:1655989312  NOTICE: [plugin.audio.radio_de] __urlopen opening url=http://players.creacast.com/creacast/hitradio_ohr/playlist.pls
20:36:11 T:1655989312  NOTICE: [plugin.audio.radio_de] __resolve_playlist found .pls file
20:36:11 T:1655989312  NOTICE: [plugin.audio.radio_de] __resolve_playlist found 6 servers
20:36:11 T:1655989312  NOTICE: [plugin.audio.radio_de] get_stream_url result: http://str0.creacast.com/hitradio_ohr
20:36:21 T:1969565696   ERROR: CCurlFile::Stat - Failed: Timeout was reached(28) for http://str0.creacast.com/hitradio_ohr
20:36:21 T:1969565696   ERROR: CAudioDecoder: Unable to Init Codec while loading file http://str0.creacast.com/hitradio_ohr
20:36:21 T:1969565696 WARNING: PAPlayer::QueueNextFileEx - Failed to create the decoder
20:36:21 T:1969565696   ERROR: Playlist Player: skipping unplayable item: 14, path [plugin://plugin.audio.radio_de/station/1623]
```

And this is with 5.0.1
```
21:26:52 T:2681508928  NOTICE: -->Python Interpreter Initialized<--
21:26:54 T:2681508928  NOTICE: [xbmcswift2] Request for "/station/2273" matches rule for function "get_stream_url"
21:26:54 T:2681508928  NOTICE: [plugin.audio.radio_de] get_station_by_station_id started with station_id=2273
21:26:54 T:2681508928  NOTICE: [plugin.audio.radio_de] __api_call started with path=broadcast/getbroadcastembedded, param={'broadcast': '2273'}
21:26:54 T:2681508928  NOTICE: [plugin.audio.radio_de] __urlopen opening url=http://radio.de/info/broadcast/getbroadcastembedded?broadcast=2273
21:26:54 T:2681508928  NOTICE: [plugin.audio.radio_de] get_stream_url result: http://swr-mp3-m-swr1bw.akacast.akamaistream.net/7/245/137133/v1/gnl.akacast.akamaistream.net/swr-mp3-m-swr1bw?mp3
21:26:54 T:2654450752  NOTICE: Thread GUIDialogCache start, auto delete: true
21:26:58 T:2785322048  NOTICE: Thread FileCache start, auto delete: false
21:26:58 T:2726396992  NOTICE: Thread ShoutcastFile start, auto delete: false
21:26:59 T:2852430912  NOTICE: Thread PAPlayer start, auto delete: false
```

I checked twice, if I had an internet / LAN connection problem, but negative.

Maybe some of the changes from 5.0.1 to 5.0.2 are incompatible with the radio addon, but the addtional problem with remote control app is puzzling me.

[1] http://kodi.wiki/view/Add-on:Radio

**Comment by fritsch** [comment]
Full debuglog please and also enable additional libcurl logging, if possible. I suspect: https://github.com/OpenELEC/OpenELEC.tv/commit/10a557abf4f212ff7d5e519fdaf1c10131738681

**Comment by MasteRehm** [comment]
I would like to provide you both debug logs, but my installed version is still 5.0.1 and I can't get the 5.0.2 tarball at the moment as you surely know ;)

**Comment by fritsch** [comment]
We will provide you a build in some hours, that you please could test for us. It includes the patch I posted reverted and a downgrade of libcurl to 7.37.1

Thanks for your help in advance.

**Comment by stefansaraev** [comment]
30min

**Comment by stefansaraev** [comment]
@MasteRehm here a (build)[http://saraev.ca/oe/OpenELEC-Generic.x86_64-devel-20150210225640-r20038-g250140a.tar] for you. could you please test and report back ?

**Comment by MasteRehm** [comment]
Oh very sorry, I didn't mention that I use a raspberry pi. Can I use the generic build?

**Comment by stefansaraev** [comment]
sure, sure, I suppose I am done with all this nonsense now..

**Comment by stefansaraev** [comment]
I will merge #3887 anyway, unlees someone says "stop! dont!" so @sraue please, meanwhile, do an rpi build with #3887 minus the curlftp fix. the world resolves around rpi, people always forget to mention they use rpi,. so it deserves one more "fast" release. thanks.

**Comment by fritsch** [comment]
Damn - this is why we have a contributing link when opening issues ... too much time wasted now :-(

**Comment by wsnipex** [comment]
I'd really like to see a debug log of 5.0.2 with curl extra logging

**Comment by fritsch** [comment]
@MasteRehm Please provide the log with curl extra debuglogging and 5.0.2

**Comment by fritsch** [comment]
@MasteRehm You can also use 5.0.3 - which also should not work: http://openelec.tv/get-openelec

**Comment by MasteRehm** [comment]
First of all I want to apologize, that I have not mentioned I'm using a raspberry pi. I use my rpi only for OpenELEC, so I forget to mention that in the first moment.

> Damn - this is why we have a contributing link when opening issues ... too much time wasted now :-(

Now I see the link. Damn, I have simply overlooked it.
OK, I'll try it again with 5.0.3 and debug logging, but not in the next 10 hours. I have to work ;)

**Comment by 1358** [comment]
Debug log with enabled CURL logging:
http://xbmclogs.com/show.php?id=422926

**Comment by fritsch** [comment]
@1358 Yeah 5.0.3 has the same curl and the same patch

Can you try:
https://dl.dropboxusercontent.com/u/55728161/OpenELEC-RPi.arm-devel-20150211075718-r20043-g4026ce3.tar <- what happens with this image?

**Comment by 1358** [comment]
It works. Thanks for your super-fast support!

for the sake of completeness: http://xbmclogs.com/show.php?id=422959

**Comment by fritsch** [comment]
https://github.com/fritsch/OpenELEC.tv/commits/oecurlrevert <- for completeness, I will redo the build but keep libcurl 7.40.0

**Comment by wsnipex** [comment]
meh, stupid shoutcast servers that simply don't return any data on HEAD requests.

**Comment by fritsch** [comment]
@1358 here is another testbuild: https://dl.dropboxusercontent.com/u/55728161/OpenELEC-RPi.arm-devel-20150211114136-r20044-g16c5d34.tar

**Comment by 1358** [comment]
@fritsch still working.
OpenELEC (unofficial) Version: devel-20150211114136-r20044-g16c5d34
OpenELEC git: 16c5d34ab417398beebd41109f6e1a40439d39e4

Do you need logs?

**Comment by fritsch** [comment]
Nope. Thanks very much. I will push our findings shortly.

**Comment by fritsch** [comment]
@stefansaraev can you pick the kodi patch revert please to your current PR concerning the libcurl downgrade?

Then we can close this bugreport here and the other with the ftp problems.

**Comment by wsnipex** [comment]
just for completeness. Its still bugged and the other patch is just bandaid. The real issue here with shoutcast is that we don't treat ICY headers correctly.

**Comment by stefansaraev** [comment]
@fritsch done. I'll push only to master when the time comes. if anyone wants to pick for 5.0 - ok. I wont.

**Comment by stefansaraev** [comment]
closing as fixed.

**Comment by fritsch** [comment]
@1358 @MasteRehm @delinend we need your help to cure kodi's curl wrapper. @wsnipex has done another approach to fix that. 
So, please check your radio streams and @delinend please check for your ftp thumbs, please.

Can you please test: https://dl.dropboxusercontent.com/u/55728161/OpenELEC-RPi.arm-devel-20150212082113-r20045-g85043f4.tar
for us?

Keep it running for some time and whenever you find a regression, where you think it's curl related - please get us a curl enabled debuglog. Thanks much in advance.

**Comment by delinend** [comment]
@fritsch : I have now tested r20045 (openELEC 5.0.3 + update to r20045), and the FTP thumbs load fine :-) :+1: 
Here is my log: http://pastebin.com/KytpF3Pm

**Comment by MasteRehm** [comment]
I installed the suggested version a moment ago.
```
OpenELEC (unofficial) Version: devel-20150212082113-r20045-g85043f4
OpenELEC git: 85043f4e72fa30751313ac843b1a443c56832fd3
```
No problem so far. Works as it should. Good work :)
I'll test it the next hours. Do you still need a debug log?

**Comment by fritsch** [comment]
Debuglog is only required when something stops working, that worked before.

Thanks much.

@snipex: That patch feels safe :-)

**Comment by Snipex** [comment]
Guten Tag,

ich bin vom 9.2.2015 bis am 18.2.2015 abwesend. Ihre Email wird nicht weitergeleitet.

&nbsp;

In dringenden F&auml;llen wenden Sie sich bitte an unsere Hotline (support@phoenix-systems.ch) oder Tel: +41 44 500 86 41

&nbsp;

--

Mit freundlichen Gr&uuml;ssen

Thomas Taroni

&nbsp;

Phoenix Systems GmbH

Rautistrasse 12

8047 Z&uuml;rich

&nbsp;

Internet: http://www.phoenix-systems.ch

Tel:&nbsp; +41 44 500 86 41

Mobile: +41 79 736 29 66

Email:&nbsp;&nbsp;&nbsp; thomas.taroni@phoenix-systems.ch

**Comment by wsnipex** [comment]
thanks for testing! Then it can go in master next window. Let me know if any problems pop up.

**Comment by cmbr** [comment]
MasteRehm did not mention which Raspberry Pi model was used for testing above. I had the same loss of radio streaming problem with my old Raspberry Pi Model B when I upgraded from OpenELEC version 5.0.0 to 5.0.3 today. When I installed the version provided by the *.tar* file linked above, the radio addon began to work again. Elsewhere (http://forum.kodi.tv/showthread.php?tid=119362&pid=1925261#pid1925261) it has been reported that the fix does not work for the Raspberry Pi Model 2 B.

**Comment by fritsch** [comment]
There was no build done for the Model 2 at all ... and that fix is generic, does not matter if running on the B or on the 2 B, it was a kodi patch.

**Comment by fritsch** [comment]
@cmbr 

> Raspberry Pi 2
RPi2: https://dl.dropboxusercontent.com/u/55728161/OpenELEC-RPi2.arm-devel-20150213173432-r20051-g9db9dda.tar 

> Raspberry Pi (the old Pi)
RPi:  https://dl.dropboxusercontent.com/u/55728161/OpenELEC-RPi.arm-devel-20150212082113-r20045-g85043f4.tar

Make sure to download the correction version if you want to help testing. Backup recommended.

**Comment by dersphere** [comment]
Ah nice, since a few weeks I get about 5 emails/week regarding playback problems with my add-ons radio and shoutcast. Now I now why, thanks for this fix!

**Comment by fritsch** [comment]
@dersphere thank @wsnipex - he fixed it.

I hope we can make another 5.0.x build with old curl (7.37.1) soon - as this does not have the issue - don't want to risk another regression again with the bumpery experiment.

If you get other mails, link them this thread, please.

**Comment by dersphere** [comment]
thanks wsnipex! (I don't want to ping you again ;-))

**Comment by fthiery** [comment]
I confirm it fixes the 2 issues (webradio timeouts and remote issues); i am using the normal built-in m3u playlist, not an extension on RPI B+, fresh install.

I noticed that yatse was failing the event server test; isn't that UDP based ? What's the relationship to curl (if any) ?

**Comment by mojocl** [comment]
Hi guys,

I don't know if my prolem is related to the same behaviour,
but I can't play any radio sender in TuneIn and radio addon.

I can reproduce it on 3 different scenarios, but I know it worked times ago.

Scenario 1: RasPi, Gen B, Openelec 5.03 & 4.2.3
Scenario 2: RasPi, Gen B+, Openelec 4.2.3
Scenario 3: RasPi 2, Openelec 5.02 & 5.03

`11:02:12 T:1969381376   ERROR: Playlist Player: skipping unplayable item: 3, path [plugin://plugin.audio.tuneinradio/?logo=http%3A%2F%2Fd1i6vahw24eb07.cloudfront.net%2Fs161194q.png&path=tune&id=s161194&name=RauteMusik.FM+Trance]
11:02:12 T:1969381376 WARNING: Attempt to set unplayable index 3`

`11:05:25 T:1969381376   ERROR: CAudioDecoder: Unable to Init Codec while loading file http://rock-high.rautemusik.fm
11:05:25 T:1969381376 WARNING: PAPlayer::QueueNextFileEx - Failed to create the decoder
11:05:25 T:1969381376   DEBUG: OnQueueNextItem: play state was 1, starting 1
11:05:25 T:1969381376   DEBUG: OnPlayBackStopped: play state was 1, starting 0
11:05:25 T:1969381376   DEBUG: CAnnouncementManager - Announcement: OnStop from xbmc
11:05:25 T:1969381376   DEBUG: GOT ANNOUNCEMENT, type: 1, from xbmc, message OnStop
11:05:25 T:1969381376   ERROR: Playlist Player: skipping unplayable item: 1, path [plugin://plugin.audio.tuneinradio/?logo=http%3A%2F%2Fd1i6vahw24eb07.cloudfront.net%2Fs153325q.png&path=tune&id=s153325&name=RauteMusik.FM+Rock]
11:05:25 T:1969381376 WARNING: Attempt to set unplayable index 1
11:05:25 T:1969381376   DEBUG: Playlist Player: one or more items failed to play... aborting playback`

I had activated Debug, but it seems that the information of addon "radio" aren't in it.

BR,

**Comment by andyjefferson** [comment]
Current TuneIn Radio plugin works with the just released OpenELEC 5.0.4.
